prov/sm2/src/sm2_coordination.c

```diff
@@ -237,8 +244,15 @@ ssize_t sm2_file_open_or_create(struct sm2_mmap *map_shared)
 	 * data exchange regions yet.
 	 */
 	err = sm2_mmap_remap(&map_ours, header->ep_regions_offset);
-	if (err)
-		return err;
+
+	if (err) {
+	early_exit:
```

Comment:
Don't love the look of this error handling. Can you modify to have a normal goto err and have the err case at the bottom with a goto err here instead? Looks kind of weird with this goto in the middle of the function and inside the if. Personal prefernce

Reply:
I moved it to the bottom.  You are right, looks better.